src/platform/packages/shared/kbn-storage-adapter/src/index_adapter/index.ts

```diff
@@ -375,28 +329,15 @@ export class StorageIndexAdapter<
     ...request
   }): Promise<StorageClientIndexResponse> => {
     const attemptIndex = async (): Promise<IndexResponse> => {
-      const [danglingItem] = id ? await this.getDanglingItems({ ids: [id] }) : [undefined];
-
-      const [indexResponse] = await Promise.all([
-        wrapEsCall(
-          this.esClient.index({
-            ...request,
-            id,
-            refresh,
-            index: this.getWriteTarget(),
-            require_alias: true,
-          })
-        ),
-        danglingItem
-          ? wrapEsCall(
-              this.esClient.delete({
-                id: danglingItem.id,
-                index: danglingItem.index,
-                refresh,
-              })
-            )
-          : Promise.resolve(),
```

Comment:
This logic has an edge case of data loss/duplication if one of the `.index` or `.delete` calls fails while the other succeeds, but since we're removing handling rollover we can remove this cleanup for now.